### DISCUSSION TITLE
void Stop()?

### DISCUSSION THREAD

**Discussion by TricksterSam** [djphazer/O_C-Phazerville#47] [Q&A]
Can you use void Stop() in an app? I don't think I've seen any HEM apps that do.

**Comment by TricksterSam** [comment]
I don't see it in the typedef struct, so I'm assuming "no". Is there another way to achieve something similar?

**Comment by djphazer** [comment]
Is this a C++ 101 thread? 😜
Or are you looking for an existing Stop() function for something?

You're free to define (or override) pretty much any function you want in an applet class, or anywhere else. Just make stuff up! Trial & Error is a great strategy: make a modification, try to compile. If it fails, deciphering the errors is where I learn valuable lessons.

I've thought about making a video tutorial about how to write applets for O_C... tips and tricks, etc.

**Reply by TricksterSam** [reply]
Thanks. I'm sorry for very basic questions. There's still a lot I'm learning. I'm doing a lot of trial and error and openai is helping me understand the syntax of C++. It's a very patient teacher but not particularly aware of anything at all. 

Was hoping I could set a variable in a header file that two apps could access, and Stop() would make it really easy to indicate when one app is no longer loaded. I'm going to keep hacking away until I figure it out, whether Stop() works or not.

**Reply by djphazer** [reply]
Ah ok, you're trying to share state between applets. This is where you want to do something similar to the HSProbLoopLinker - the best way to know if an applet is still loaded is to have it "register" itself every time Controller() is called, and have a function to check whether the registration tick is recent enough. There aren't really any existing mechanisms that are triggered when "closing" or switching away from an applet...

**Reply by TricksterSam** [reply]
Thanks. I had been looking at HSRingBuffer because I knew ASR used it and that app links. ProbabilityDivider looks more straight forward.

**Reply by djphazer** [reply]
Yep, same concept. The Lorenz generator manager also works the same way.

**Reply by TricksterSam** [reply]
Thank you for your time!

**Reply by TricksterSam** [reply]
Finally, got it working right. Thank you. Now to deleting a bunch of commented code.
